src/menu.scss

```diff
@@ -5,165 +5,295 @@
+      // for desktop mode
+      padding: 0.375rem 0.75rem 0.375rem 0.5rem; // line-height=20px so 2rem or 32px-20px=12px/2= 6px or 0.375rem each
+      height: 2rem;
+    }
+
+    .#{$block}__addon-before {
+      width: 2rem; // to account for item right padding of 12px or 0.75rem so 2rem-0.75rem = 1.25rem
+      height: 1.25rem; // to account for item top padding of 6px or 0.375rem so 2rem-0.375rem-0.375rem = 1.25rem
+      margin: 0 0 0 -0.5rem;
+    }
+
+    .#{$block}__addon-after {
+      width: 2rem; // to account for item right padding of 12px or 0.75rem so 2rem-0.75rem = 1.25rem
+      height: 1.25rem; // to account for item top padding of 6px or 0.375rem so 2rem-0.375rem-0.375rem = 1.25rem
+      margin: 0 -0.75rem 0 0;
+    }
+
+    // handle icons rtl modes
+    @include fd-rtl() {
+      .#{$block}__addon-before {
+        margin: 0 -0.75rem 0 0; // since before and after icons switch places, their margins also switch
+      }
+
+      .#{$block}__addon-after {
+        margin: 0 0 0 -0.5rem; // since before and after icons switch places, their margins also switch
       }
     }
   }
 
-  &__title {
-    display: block;
-    margin-bottom: 0;
-    margin-top: 0;
-    padding: $fd-menu-title-padding $fd-menu-item-padding-x;
+  &--mobile {      // for mobile mode
+    @include fd-reset();
+
+    max-width: none;
+
+    .#{$block}__item {
+      padding: 0.75rem 1rem 0.75rem 1rem; // line-height=20px so 2.75rem or 44px-20px=24px/2= 12px or 0.75rem each
+      height: 2.75rem;
+      width: 100%;
+      max-width: none;
+
+      .#{$block}__title {
+        max-width: 100%;
+      }
+    }
+
+    .#{$block}__addon-before {
+      width: 2.75rem; // to account for item right padding of 16px or 1rem so 2.75rem-1rem = 1.75rem
+      height: 1.25rem; // to account for item top padding of 12px or 0.75rem so 2.75rem-0.75rem-0.75rem = 1.25rem
+      margin: 0 0 0 -1rem;
+    }
 
-    @include fd-type("-1", "normal");
+    .#{$block}__addon-after {
+      width: 2.75rem; // to account for item right padding of 16px or 1rem so 2.75rem-1rem = 1.75rem
+      height: 1.25rem; // to account for item top padding of 12px or 0.75rem so 2.75rem-0.75rem-0.75rem = 1.25rem
+      margin: 0 -1rem 0 0;
+    }
 
-    text-transform: uppercase;
+    @include fd-rtl() {
+      .#{$block}__addon-before {
+        margin: 0 -1rem 0 0; // since before and after icons switch places, their margins also switch
+      }
 
-    @include fd-var-color("color", $fd-menu-title-color, --fd-menu-title-color);
-    @include fd-var-color("background-color", $fd-menu-title-background-color, --fd-menu-title-background-color);
+      .#{$block}__addon-after {
+        margin: 0 0 0 -1rem; // since before and after icons switch places, their margins also switch
+      }
+    }
   }
 
   &__item {
     @include fd-reset();
-    @include fd-ellipsis();
 
-    display: block;
+    display: flex;
+    align-items: center;
     transition: all $fd-animation-speed ease-in;
```

Comment:
There is no mention about transition in menu specs

Reply:
Missed this. Will remove it, thanks for pointing it out!